Fix GraphQL patchers 'require' issue

vpellan opened the pull request:
<!--
Check out the
https://github.com/DataDog/dd-trace-rb/blob/master/docs/DevelopmentGuide.md
for guidance on how to set up your development environment,
run the test suite, write new integrations, and more.
-->

**What does this PR do?**
<!-- A brief description of the change being made with this pull request. -->
Making a custom tracer based on unified_tracer should now not need to require explicitly the unified_tracer.rb file anymore. This also fixes the problem with currently unreleased appsec_tracer.

**Motivation:**
<!-- What inspired you to submit this pull request? -->
Fixes issue #3804 

**Additional Notes:**
<!-- Anything else we should know when reviewing? -->

**How to test the change?**
<!--
Describe here how the change can be validated.
You are strongly encouraged to provide automated tests for this PR (unit or integration).
If this change cannot be feasibly tested, please explain why,
unless the change does not modify code (e.g. only modifies docs, comments).
-->

Create a Rails + GraphQL app with a custom tracer that includes unified_tracer. GraphQL AppSec test app is using this so if system-tests passes, it works.

Unsure? Have a question? Request a review!

p-datadog commented:
Is it supported to load datadog first and graphql second, as well as the other way around?

We already have tests like https://github.com/DataDog/dd-trace-rb/blob/master/spec/loading_spec.rb that start with a clean environment, given that this was a customer-reported issue I would consider adding a test that verifies that the reported case works correctly going forward.

vpellan commented:
> Is it supported to load datadog first and graphql second, as well as the other way around?
> 
> We already have tests like [`master`/spec/loading_spec.rb](https://github.com/DataDog/dd-trace-rb/blob/master/spec/loading_spec.rb?rgh-link-date=2024-07-30T17%3A50%3A46Z) that start with a clean environment, given that this was a customer-reported issue I would consider adding a test that verifies that the reported case works correctly going forward.

I've added a test that will require datadog first, then graphql, and the other way around too.
I've also added a test for trace_with.

p-datadog commented on spec/datadog/tracing/contrib/graphql/loading_spec.rb:
```diff
@@ -0,0 +1,31 @@
+RSpec.describe 'loading graphql' do
```
I would think that adding `require 'shellwords'` in this file is necessary so that it can be executed by itself? The only other place that requires this library is the top-level loading test that I can see.

vpellan replied:
Thanks for the suggestion, I've added it !
